File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> ust;
        for(int i=0; i<(int)nums.size(); ++i){
            ust.insert(nums[i]);
        }
        int maxLen=0, currLen;
        for(int a: ust){
            if(ust.find(a-1)!=ust.end()) continue;
            currLen=1;
            while(ust.find(a+1)!=ust.end()){
                ++currLen;
                ++a;
            }
            maxLen=max(maxLen, currLen);
        }
        return maxLen;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> cache= new HashSet<>();
        for(int x: nums) cache.add(x);
        int ans= 0;
        for(int x: cache){
            if(cache.contains(x-1)) continue;  // ensures we only start from the beginning of a sequence
            int val= x, count= 0;
            while(cache.contains(val)){
                count++;
                val++;
            }
            if(ans<count) ans= count;
        }
        return ans;
    }
}
